File: 2024/02/wxg__two-largest-number/1.py
# 给定n个正整数，求选出两个数拼接在一起能得到的最大整数 （ 例如1和2拼接得到21，123、4、5拼接得5123）

if __name__ == '__main__':
    arr = [101, 100, 100]
    sorted_arr = sorted(arr, reverse=True)
    res1 = f'{sorted_arr[0]}{sorted_arr[1]}'
    res2 = f'{sorted_arr[1]}{sorted_arr[0]}'
    print(res1 if int(res1) > int(res2) else res2)

File: 2024/02/validate-binary-search-tree/1.js
// https://leetcode.cn/problems/validate-binary-search-tree/description/

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

function TreeNode(val, left, right) {
    this.val = (val === undefined ? 0 : val)
    this.left = (left === undefined ? null : left)
    this.right = (right === undefined ? null : right)
}


/**
 * @param {TreeNode|null} root
 * @return {boolean}
 */
var isValidBST = function (root) {
    /**
     *
     * @type {number[]}
     */
    const arr = [];

    /**
     * @param {TreeNode|null} root
     * @return {boolean}
     */
    var helper = function (root) {
        if (!root) {
            return true;
        }
        const res1 = helper(root.left);

        if (arr.length && arr[arr.length - 1] >= root.val) {
            return false;
        }
        arr.push(root.val);

        const res2 = helper(root.right);
        return res1 && res2;
    };

    return helper(root);
};